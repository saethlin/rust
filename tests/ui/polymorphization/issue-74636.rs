//@ compile-flags:-Zpolymorphize=on
//@ build-pass
//@ dont-check-compiler-stderr

use std::any::TypeId;

pub fn foo<T: 'static>(_: T) -> TypeId {
    TypeId::of::<T>()
}

fn outer<T: 'static>() {
    foo(|| ());
}

fn main() {
    outer::<u8>();
}
