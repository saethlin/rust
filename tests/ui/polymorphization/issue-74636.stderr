 WARN rustc_codegen_ssa::mir::locals Unexpected initial operand type:
 expected Closure(DefId(0:8 ~ issue_74636[7d87]::outer::{closure#0}), [Closure(DefId(0:8 ~ issue_74636[7d87]::outer::{closure#0}), [T/#0, i8, Binder { value: extern "RustCall" fn(()), bound_vars: [] }, ()]), i8, Binder { value: extern "RustCall" fn(()), bound_vars: [] }, ()]),
 found    Closure(DefId(0:8 ~ issue_74636[7d87]::outer::{closure#0}), [T/#0, i8, Binder { value: extern "RustCall" fn(()), bound_vars: [] }, ()]).
 See <https://github.com/rust-lang/rust/issues/114858>.
